src/queue/PlaylistState.cxx

```diff
@@ -152,7 +154,7 @@ playlist_state_restore(const char *line, TextFile &file,
 		} else if ((p = StringAfterPrefix(line, PLAYLIST_STATE_FILE_REPEAT))) {
 			playlist.SetRepeat(pc, StringIsEqual(p, "1"));
 		} else if ((p = StringAfterPrefix(line, PLAYLIST_STATE_FILE_SINGLE))) {
-			playlist.SetSingle(pc, StringIsEqual(p, "1"));
+			playlist.SetSingle(pc, (SingleMode)atoi(p));
```

Comment:
This cast can cause invalid values to be set, e.g. if a state file from a future MPD version happens to get loaded. The state file loader should be robust against such a thing.